### PR TITLE
Introduce ConfirmedMigrating wallet state

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      email: this.email,
+      ensName: this.ensName,
+      walletJSON: JSON.parse(await wallet.encrypt(password)),
+      contractAddress: this.contractAddress,
+      publicKey: wallet.address,
+    };
+    const {email} = await this.sdk.relayerApi.storeEncryptedWallet(storedEncryptedWallet, this.code);
+    if (email === this.email) {
+      return new DeployedWallet(this.contractAddress, this.ensName, this.privateKey, this.sdk, this.email);
+    }
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts:40]: We can throw an error here e.g. `Enable to set password`

File: universal-login-sdk/test/api/wallet/ConfirmedMigratingWallet.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      contractAddress: confirmedWallet.contractAddress,
+      code: confirmedWallet.code,
+    });
+  });
+
+  it('roundtrip', async () => {
+    const password = 'SuperStrongNewPassword';
+    const result: any = await confirmedWallet.setPassword(password);
+    expect(result instanceof DeployedWallet).to.be.true;
+  });
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/test/api/wallet/ConfirmedMigratingWallet.int.test.ts:42]: I would add flow with restoring wallet here to be sure that the encrypted wallet is stored in the relayer.

File: universal-login-commons/src/core/models/wallets/SerializableConfirmedMigratingWallet.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import {asString, asObject} from '@restless/sanitizers';
+
+export interface SerializableConfirmedMigratingWallet {
```
Comment (xitronix) [inline universal-login-commons/src/core/models/wallets/SerializableConfirmedMigratingWallet.ts:3]: I think we should extend ConfirmedWallet here. What do you think?

File: universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts (hunk 1 above)
Comment (xitronix) [reply universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts:40]: Is it possible to `email !== this.email`?

File: universal-login-commons/src/core/models/wallets/SerializableConfirmedMigratingWallet.ts (hunk 3 above)
Comment (JustynaBroniszewska) [reply universal-login-commons/src/core/models/wallets/SerializableConfirmedMigratingWallet.ts:3]: We could but it is more readable this way.

File: universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts (hunk 1 above)
Comment (JustynaBroniszewska) [reply universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts:40]: We could do `ensure(email === this.email, CreatingPasswordFailed)`, what do you think?

Comment (xitronix) [reply universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts:40]: Good idea

**Review by xitronix — APPROVED** [review]

File: universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import {Wallet} from 'ethers';
+import {SerializableConfirmedMigratingWallet, StoredEncryptedWallet, ensure} from '@unilogin/commons';
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/src/api/wallet/ConfirmedMigratingWallet.ts:2]:
```suggestion
import {SerializableConfirmedMigratingWallet, StoredEncryptedWallet} from '@unilogin/commons';
```

**Review by xitronix — APPROVED** [review]

**Pull request merged**